commands/create/nodepool/command.go

```diff
@@ -0,0 +1,357 @@
+- Instance type used
+
+When creating a node pool, all arguments (except for the cluster ID) are
+optional. Where an argument is not given, a default will be applied as
+follows:
+
+- Name: will be "Unnamed node pool <n>".
+- Availability zones: the node pool will use 1 zone selected randomly.
+- Instance type: the default instance type of the installation will be
+  used. Check 'gsctl info' to find out what that is.
+- Scaling settings: the minimum and maximum size will be set to 3,
+  meaning that autoscaling is disabled.
+
+Examples:
+
+  The simplest invocation creates a node pool with default attributes
+  in the cluster specified.
+
+    gsctl create nodepool f01r4
+
+  We recommend to set a descriptive name, to tell the node pool apart
+  from others.
+
+    gsctl create nodepool f01r4  --name "Batch jobs"
+
+  Assigning the node pool to availabilty zones can be done in several
+  ways. If you only want to ensure that several zones are used, specify
+  a number liker like this:
+
+    gsctl create nodepool f01r4 --num-availability-zones 2
+
+  To set one or several specific zones to use, give a list of zone names
+  or letters.
+
+    gsctl create nodepool f01r4 --availability-zones b,c,d
+
+  Here is how you specify the instance type to use:
+
+    gsctl create nodepool f01r4 --aws-instance-type m4.2xlarge
+
+  The initial node pool size is set by adjusting the lower and upper
+  size limit like this:
+
+    gsctl create nodepool f01r4 --nodes-min 3 --nodes-max 10
+`,
+
+		// PreRun checks a few general things, like authentication.
+		PreRun: printValidation,
+
+		// Run calls the business function and prints results and errors.
+		Run: printResult,
+	}
+
+	cmdAwsEc2InstanceType   string
+	cmdName                 string
+	cmdAvailabilityZonesNum int
+	cmdAvailabilityZones    []string
+)
+
+const (
+	activityName = "create-nodepool"
+)
+
+func init() {
+	initFlags()
+}
+
+// initFlags initializes flags in a re-usable way, so we can call it from multiple tests.
+func initFlags() {
+	Command.ResetFlags()
+	Command.Flags().StringVarP(&cmdName, "name", "n", "", "name or purpose description of the node pool")
+	Command.Flags().IntVarP(&cmdAvailabilityZonesNum, "num-availability-zones", "", 0, "Number of availability zones to use. Default is 1.")
+	Command.Flags().StringSliceVarP(&cmdAvailabilityZones, "availability-zones", "", nil, "List of availability zones to use, instead of setting a number. Use comma to separate values.")
+	Command.Flags().StringVarP(&flags.CmdWorkerAwsEc2InstanceType, "aws-instance-type", "", "", "EC2 instance type to use for workers, e. g. 'm5.2xlarge'")
+	Command.Flags().Int64VarP(&flags.CmdWorkersMin, "nodes-min", "", 0, "Minimum number of worker nodes for the node pool.")
+	Command.Flags().Int64VarP(&flags.CmdWorkersMax, "nodes-max", "", 0, "Maximum number of worker nodes for the node pool.")
+}
+
+// Arguments defines the arguments this command can take into consideration.
+type Arguments struct {
```

Comment:
For the future, when we align command line tools more then this would look differently. In SIG Operator we try to align command line tools more to https://github.com/giantswarm/devctl.